QuasistaticBrownianThermalNoise/QuasistaticBrownianThermalNoise.cpp

```diff
@@ -22,1587 +22,1671 @@
-  }
+  const unsigned int n_points = points.size();
 
-  //****************************************************************//
-  // ElasticProblem implementation
-  //****************************************************************//
-  template <int dim> double
-  ElasticProblem<dim>::getYijkl(int whichYijkl, unsigned int i, 
-				unsigned int j, unsigned int k, 
-				unsigned int l) {
-    //Choices are 
-    //kAlGaAs #crystal, x=0.92
-    //kIso_AlGaAs #effective isotropic approx to x=0.92 AlGaAs
-    //kIso_Ta2O5 #tantalum
-    //kIso_FusedSilica #fused silica
-    switch(whichYijkl) {
-    case(kAlGaAs): 
+  for (unsigned int p = 0; p < n_points; ++p)
+    RightHandSide<dim>::vector_value(points[p], value_list[p]);
+}
+
+//****************************************************************//
+// ElasticProblem implementation
+//****************************************************************//
+template <int dim>
+double ElasticProblem<dim>::getYijkl(int whichYijkl, unsigned int i,
+                                     unsigned int j, unsigned int k,
+                                     unsigned int l) {
+  // Choices are
+  // kAlGaAs #crystal, x=0.92
+  // kIso_AlGaAs #effective isotropic approx to x=0.92 AlGaAs
+  // kIso_Ta2O5 #tantalum
+  // kIso_FusedSilica #fused silica
+  switch (whichYijkl) {
+    case (kAlGaAs):
       return Y_AlGaAs[i][j][k][l];
-    case(kIso_AlGaAs): 
+    case (kIso_AlGaAs):
       return Y_Iso_AlGaAs[i][j][k][l];
-    case(kIso_Ta2O5):
+    case (kIso_Ta2O5):
       return Y_Iso_Ta2O5[i][j][k][l];
-    case(kIso_FusedSilica):
+    case (kIso_FusedSilica):
       return Y_Iso_FusedSilica[i][j][k][l];
     default:
       std::cout << "WARNING! Invalid Yijkl string. Do not trust results!\n";
       break;
+  }
+  return 0.0;
+}
+
+// Switch case to find Yijkl using more than one loss angle for the coating.
+template <int dim>
+double ElasticProblem<dim>::getMultLossAngleYijkl(int whichYijkl,
+                                                  unsigned int i,
+                                                  unsigned int j,
+                                                  unsigned int k,
+                                                  unsigned int l) {
+  switch (whichYijkl) {
+    case (kAlGaAs):
+      return Y_AlGaAs_Phi[i][j][k][l];
+    case (kIso_AlGaAs):
+      return Y_Iso_AlGaAs_Phi[i][j][k][l];
+    case (kIso_Ta2O5):
+      return Y_Iso_Ta2O5_Phi[i][j][k][l];
+    case (kIso_FusedSilica):
+      return Y_Iso_FusedSilica_Phi[i][j][k][l];
+    default:
+      std::cout << "WARNING! Invalid Yijkl string. Do not trust results!\n";
+      break;
+  }
+  return 0.0;
+}
+
+// Calculates and returns the correct value for Y_ijkl in terms of the bulk
+// and shear modulus. The ints i,j,k,l that are passed are the ints from the for
+// loops that assign the correct values to the youngs tensor for each of the
+// coatings.
+
+double ReturnBulkValue(double lame_mu, double lame_lambda, double phibulk,
```

Comment:
needs a better name and a comment. What does this function do?

It looks like this term figures out certain Yijkl components...where do these equations come from?